produce_snp_matrix.py

```diff
@@ -1,46 +1,86 @@
 from glob import glob
-import subprocess
+import json
 import argparse
 import utils
 import os
 
-def snps(input_path, output_path):
+def download_test_case(case, download_path):
+    with open('./test.json') as f:
+        x =json.load(f)
+
+        try:
+            y = x[case]
+        except:
+            print(f"no case named {case}")
+        
+        num_samples = len(y)
+
+        while num_samples > 0:
+            sample = y[str(num_samples)]
+            r1_uri = sample["r1_uri"]
+            r2_uri = sample["r2_uri"]
+
+            if os.path.exists(download_path + os.path.basename(r1_uri)):
+                print("read already downloaded, skipping")
+            else:
+                cmd = f"aws s3 cp --dryrun {r1_uri} {download_path}"
+                utils.run(cmd, shell=True)
+
+            if os.path.exists(download_path + os.path.basename(r2_uri)):
+                print("read already downloaded, skipping")
+            else:
+                cmd = f"aws s3 cp --dryrun {r2_uri} {download_path}"
+                utils.run(cmd, shell=True)
+            num_samples = num_samples - 1
+
+
+def btb_seq(download_path, output_path):
```

Comment:
To avoid hard coding the btb-seq path in line 39, you could pass it in as an argument. See the function `sequence()` in `validator.py`.  I'm wondering if you could just either copy the whole code for this function or maybe better still, just import it into this file and use it here ...